prime-react/src/main/scala/react/primereact/Button.scala

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package react.primereact
+
+import cats.syntax.all.*
+import japgolly.scalajs.react.*
+import japgolly.scalajs.react.vdom.html_<^.*
+import react.common.*
+import reactST.primereact.components.{Button => CButton}
+import reactST.primereact.primereactStrings.button
+import reactST.primereact.primereactStrings.reset
+import reactST.primereact.primereactStrings.submit
+
+import scalajs.js
+import scalajs.js.JSConverters.*
+
+object Button {
+  enum Size(val cls: Css):
+    case Small  extends Size(PrimeStyles.ButtonSmall)
+    case Normal extends Size(PrimeStyles.ButtonNormal)
+    case Large  extends Size(PrimeStyles.ButtonLarge)
+
+  enum Severity(val cls: Css):
+    case Primary   extends Severity(PrimeStyles.ButtonPrimary)
+    case Secondary extends Severity(PrimeStyles.ButtonSecondary)
+    case Success   extends Severity(PrimeStyles.ButtonSuccess)
+    case Info      extends Severity(PrimeStyles.ButtonInfo)
+    case Warning   extends Severity(PrimeStyles.ButtonWarning)
+    case Help      extends Severity(PrimeStyles.ButtonHelp)
+    case Danger    extends Severity(PrimeStyles.ButtonDanger)
+
+  enum Type(val value: submit | reset | button):
+    case Button extends Type(button)
+    case Submit extends Type(submit)
+    case Reset  extends Type(reset)
+
+  def apply(
```

Comment:
I think we should settle for either a fully "apply" or fully "builder" constructor.

We can add our own parameters or variants to the builders via extension methods.

As an example:
``` scala
extension (builder: Button.Builder)
  inline def clazz(v: Css): Button.Builder = 
    builder.className(v.htmlClass)
```

And yet:
- The underlying `StBuildingComponent` has an `.apply(TagMod)` method, so we don't actually need the `clazz: Css` properties; we can do `Button()(SomeCss)`. (Come think if it, we could have done this for Semantic too).
- I'm a bit worried about having different properties that affect the same underlying property (in this case `className`). With the builder pattern, nothing stops us from calling `.className` later on, which wipes whatever we set before. For example: 
``` scala
  Button(size = Size.Small).className("foo")
```
would result in the size being ignored.

I don't see a way around this other than writing our own facades.